File: src/main/java/at/hrechny/predictionsbot/mapper/YearMapper.java
package at.hrechny.predictionsbot.mapper;

import java.time.Year;
import org.apache.commons.lang3.StringUtils;
import org.springframework.stereotype.Service;

@Service
public class YearMapper {

  public String asString(Year year) {
    return year != null ? year.toString() : null;
  }

  public Year asYear(String string) {
    if (StringUtils.isBlank(string)) {
      return null;
    }

    return Year.parse(string);
  }

}

File: src/main/kotlin/at/hrechny/predictionsbot/mapper/YearMapper.kt
package at.hrechny.predictionsbot.mapper

import jakarta.inject.Singleton
import java.time.Year
import org.apache.commons.lang3.StringUtils

@Singleton
class YearMapper {
    fun asString(year: Year?): String? = year?.toString()

    fun asYear(string: String?): Year? =
        if (StringUtils.isBlank(string)) {
            null
        } else {
            Year.parse(string)
        }
}
